std/container/binaryheap.d

```diff
@@ -115,11 +115,26 @@ if (isRandomAccessRange!(Store) || isRandomAccessRange!(typeof(Store.init[])))
     {
         assert(!store.empty, "Cannot pop an empty store.");
         if (store.length == 1) return;
-        auto t1 = moveFront(store[]);
-        auto t2 = moveBack(store[]);
+        static if(isRandomAccessRange!Store)
+        {
+            auto t1 = moveFront(store);
```

Comment:
Perhaps instead of using `store` everywhere, add an internal primitive like:

```d
static if (isRandomAccessRange!Store)
    alias range = store;
else
    auto range() @property
    {
        return store[];
    }
```
Then use `range` everywhere. However, `BinaryHeap` clearly needs more tests too, and at this point, it would probably be better to fix it in a separate PR. It also has a strange thing going on where it uses the `package(std)` symbol `HeapOps` from std.algorithm.sorting.